### ISSUE TITLE
Allow interaction from/to emacs?

### ISSUE THREAD

**Issue by indigoviolet** [atlas-engineer/nyxt#1315]
Thanks for your work on Nyxt, I'm an enthusiastic supporter/sponsor of this project.

**tl;dr**: Have you considered allowing Nyxt to have an API to/from Emacs?

## More detail:

One of the reasons I originally became excited about Nyxt is that Emacs lacks a
decent web browser, which prevents it from smoothly integrated with many
workflows that interact with web pages.

For example, while there are tools like [Atomic Chrome](https://github.com/alpha22jp/atomic-chrome),
[Grasp](https://github.com/karlicoss/grasp) etc that allow webpages (really, browser extensions) to talk to Emacs, these are always somewhat clunky and aren't as powerful as they could be.

If Emacs had the ability to programmatically drive a browser, and receive
information back from it, I could see a number of integrations developing, for
example web annotation via org-mode, similar to tools like
[org-noter](https://github.com/weirdNox/org-noter).

Another direction could be driving web searches or documentation searches from
programming modes and pulling information back from the webpage, or embedding it
in a popup.

Currently my impression of Nyxt's direction is that it is in the family of
keyboard-driven browser interfaces like Qutebrowser, Vimium etc., which is
great - I'm an enthusiastic user of Vimium as well.

But I feel that providing an interface to the Emacs ecosystem will open up a lot
of other powerful directions. I'm not a Lisp expert, but I imagine that Nyxt being programmable in Lisp should make this sort of API more feasible/natural.

Has this been considered previously? I would love to read more if so.

Thanks again for reading this wishlist :)

**Comment by Ambrevar** [comment]
Thanks for your enthusiasm, and thanks for the kind words!

There is no official Emacs API at the moment, but it could very well
become an extension!  See

    https://nyxt.atlas.engineer/article/nyxt-extensions-tutorial.org

for a tutorial.

A while ago I wrote this article:

    https://nyxt.atlas.engineer/article/emacs-hacks.org

It's a bit dated and there are much better ways to do it, but that
should wet your appetite :)

Contributions welcome!
